app/src/org/commcare/android/tasks/TemplatePrinterTask.java

```diff
@@ -0,0 +1,253 @@
+package org.commcare.android.tasks;
+
+import android.os.AsyncTask;
+import android.os.Bundle;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.commcare.android.util.TemplatePrinterUtils;
+
+
+/**
+ * Asynchronous task for populating an html document with data.
+ *
+ * @author Richard Lu
+ * @author amstone
+ */
+public class TemplatePrinterTask extends AsyncTask<Void, Void, Integer> {
+
+    /**
+     * The 4 result codes that can be sent back by this task
+     */
+    public static final int SUCCESS = 0;
+    public static final int IO_ERROR = 1;
+    public static final int VALIDATION_ERROR_MUSTACHE = 2;
+    public static final int VALIDATION_ERROR_CHEVRON = 3;
+
+    /**
+     * Used to track which type of error (2 or 3 above) was encountered in validateString
+     */
+    private static int validationErrorType;
+
+    /**
+     * Used to track the string in the template file where a validation error was encountered
+     */
+    private String problemString;
+
+    /**
+     * The template file for this print action
+     */
+    private final File inputFile;
+
+    /**
+     * The path where the populated template should be saved to
+     */
+    private final String outputPath;
```

Comment:
I'm a fan of using self-documenting names that make comments unnecessary. Do what you prefer, but maybe consider something like 'inputFile' -> 'templateFile' and 'outputPath' -> 'populatedTemplateFilePath'